test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/ViewFeatures/CachedExpressionCompilerTest.cs

```diff
@@ -0,0 +1,835 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    public class CachedExpressionCompilerTest
```

Comment:
Add tests demonstrating the `CachedExpressionCompiler` actually caches when called multiple times with equal `Expression`s.